src/codegen.cpp

```diff
@@ -5640,6 +5625,25 @@ static void init_julia_llvm_env(Module *m)
                                      "jl_float_temp"));
 #endif
 
+#ifndef JULIA_ENABLE_THREADING
+    jltls_states_var =
+        new GlobalVariable(*m, ArrayType::get(T_int8, sizeof(jl_tls_states_t)),
```

Comment:
I think this can be declared as an ArrayType of `T_pint8` so that alignment is correct and to avoid the BitCastOp on most usages